storage/remote/client.go

```diff
@@ -145,41 +149,16 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("User-Agent", userAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
-	ctx, cancel := context.WithTimeout(ctx, c.timeout)
-	defer cancel()
-
-	httpResp, err := c.client.Do(httpReq.WithContext(ctx))
+	httpResp, err := c.client.Do(httpReq)
```

Comment:
Actually I don't encapsulate the httpReq with ctx. Because we need to keep the http connection alive, the ctx will be closed before we finished read all series.